compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIELowerToAIE.cpp

```diff
@@ -222,6 +222,70 @@ LogicalResult coreLinalgOpToAIE(IRRewriter &rewriter, linalg::LinalgOp linalgOp,
   return success();
 }
 
+LogicalResult coreMemrefExtractStridedMetadataToAIE(
+    IRRewriter &rewriter,
+    memref::ExtractStridedMetadataOp extractStridedMetadataOp,
+    IRMapping &mapper, SmallVector<Operation *> &toBeErased) {
+  LLVM_DEBUG(llvm::dbgs() << "Convert [memref.extract_strided_metadata]\n");
+  OpBuilder::InsertionGuard guard(rewriter);
+  rewriter.setInsertionPoint(extractStridedMetadataOp);
+  Value newSource =
+      mapper.lookupOrDefault(extractStridedMetadataOp.getSource());
+  memref::ExtractStridedMetadataOp newExtractStridedMetadataOp =
+      rewriter.create<memref::ExtractStridedMetadataOp>(
+          extractStridedMetadataOp.getLoc(), newSource);
+  // Map old op to new op.
+  rewriter.replaceAllUsesWith(extractStridedMetadataOp->getResults(),
+                              newExtractStridedMetadataOp->getResults());
+  toBeErased.push_back(extractStridedMetadataOp);
+  return success();
+}
+
+LogicalResult coreFuncCallOpToAIE(IRRewriter &rewriter, func::CallOp oldCallOp,
+                                  IRMapping &mapper,
+                                  SmallVector<Operation *> &toBeErased) {
+  LLVM_DEBUG(llvm::dbgs() << "Convert [func.call / function declaration]\n");
+  // Form new argument(s) and function type for the func.call op.
+  SmallVector<Value> newArgs;
+  SmallVector<Type> newArgTypes;
+  SmallVector<Type> newResultTypes;
+  for (Value operand : oldCallOp.getOperands()) {
+    Value newOperand = mapper.lookupOrDefault(operand);
+    newArgs.push_back(newOperand);
+    newArgTypes.push_back(newOperand.getType());
+  }
```

Comment:
Use `remapOperands` utility instead?

Reply:
I cannot seem to do that because I need to create a new `FuncOp` instead and attach that to the new symbol (func declaration) from the SymbolTable.